database/sqlite/sqlite_health.c

```diff
@@ -0,0 +1,1367 @@
+void sql_health_alarm_log_load(RRDHOST *host) {
+    sqlite3_stmt *res = NULL;
+    int rc;
+    char command[MAX_HEALTH_SQL_SIZE + 1];
+
+    host->health_log_entries_written = 0;
+
+    if (unlikely(!db_meta)) {
+        if (default_rrd_memory_mode == RRD_MEMORY_MODE_DBENGINE)
+            error_report("HEALTH [%s]: Database has not been initialized", host->hostname);
+        return;
+    }
+
+    char uuid_str[GUID_LEN + 1];
+    uuid_unparse_lower(host->host_uuid, uuid_str);
+    uuid_str[8] = '_';
+    uuid_str[13] = '_';
+    uuid_str[18] = '_';
+    uuid_str[23] = '_';
+
+    snprintfz(command, MAX_HEALTH_SQL_SIZE, SQL_LOAD_HEALTH_LOG(uuid_str, host->health_log.max));
+
+    rc = sqlite3_prepare_v2(db_meta, command, -1, &res, 0);
+    if (unlikely(rc != SQLITE_OK)) {
+        error_report("HEALTH [%s]: Failed to prepare sql statement to load health log.", host->hostname);
+        return;
+    }
+
+    netdata_rwlock_rdlock(&host->health_log.alarm_log_rwlock);
+
+    while (sqlite3_step(res) == SQLITE_ROW) {
+        errno = 0;
+        ssize_t errored = 0;
+
+        ALARM_ENTRY *ae = NULL;
+
+        // check that we have valid ids
+        uint32_t unique_id = sqlite3_column_int(res, 1);
+        if(!unique_id) {
+            error_report("HEALTH [%s]: Got invalid unique id. Ignoring it.", host->hostname);
+            errored++;
+            continue;
+        }
+
+        uint32_t alarm_id = sqlite3_column_int(res, 2);
+        if(!alarm_id) {
+            error_report("HEALTH [%s]: Got invalid alarm id. Ignoring it.", host->hostname);
+            errored++;
+            continue;
+        }
+
+        // Check if we got last_repeat field
+        time_t last_repeat = 0;
+        char* alarm_name = strdupz((char *) sqlite3_column_text(res, 14));
+        last_repeat = (time_t)sqlite3_column_int(res, 27);
+
+        RRDCALC *rc = alarm_max_last_repeat(host, alarm_name,simple_hash(alarm_name));
+        if (!rc) {
+            for(rc = host->alarms; rc ; rc = rc->next) {
+                RRDCALC *rdcmp  = (RRDCALC *) avl_insert_lock(&(host)->alarms_idx_name, (avl_t *)rc);
+                if(rdcmp != rc) {
+                    error("Cannot insert the alarm index ID using log %s", rc->name);
+                }
+            }
+
+            rc = alarm_max_last_repeat(host, alarm_name,simple_hash(alarm_name));
+        }
+
+        if(unlikely(rc)) {
+            if (rrdcalc_isrepeating(rc)) {
+                rc->last_repeat = last_repeat;
+                // We iterate through repeating alarm entries only to
+                // find the latest last_repeat timestamp. Otherwise,
+                // there is no need to keep them in memory.
+                continue;
+            }
+        }
+
+        freez(alarm_name);
+
+        // check for a possible host mismatch
+        if(strcmp((char *) sqlite3_column_text(res, 0), host->hostname)) {
```

Comment:
Should this be done earlier?  Maybe line 728? To avoid code in L734-L744

Reply:
I don't know if it's even needed at all.... it is also disabled in old health log code... I will remove....

Reply:
Removed in https://github.com/netdata/netdata/pull/11399/commits/9aa506af892add8c4d87681abc9040e0ee6f3413